### ISSUE TITLE
overplotting, changing styles etc.

### ISSUE THREAD

**Issue by rbiswas4** [healpy/healpy#278]
I am interested in plotting healpixels and other healpix quantities. The healpy method I found was healpy.mollview(), with an option of plotting specific pixels by masking all others.

I would like to be able to do this overplotting with other things (non-healpix), (and also change the style of plots since the particular opaque sets don't play well with overplotting). This is hard to do, since the return of mollview is `None`, so I am not sure how to proceed. Is there a different function in healpy that I should be trying to use?

It would help if 
- mollview actually returned a figure object that could be manipulated externally. That could help in solving the above problem (if there is no other function), but wold be really useful in general
- Also on my wishlist is some kind of representation of individual healpixels, in terms of boundaries and as a function of healpixId, nside and nesting scheme, that can be easily plotted in multiple ways, and will also enable solving the above problem. Is there something in healpix that provides this?

**Comment by zonca** [comment]
* you can try with the keyword `return_projected_map`, if you instead want the figure, I wonder if `gcf` function of matplotlib would work.
* about the second item, is what you need available in the IDL version? http://healpix.jpl.nasa.gov/html/idl.htm

**Comment by zonca** [comment]
no followup, closing this, reopen if needed

**Comment by rbiswas4** [comment]
Sorry, I lost track of this but I would like to return to the first point:


- I have a set of points (ra, dec) and I can convert them to theta, phi in the range [0, np.pi), [0, 2*np.pi)
- I can find the ids of the healpixels (in say the NEST scheme) these points fall on 

Now, I would like to plot a particular healpixel say healpixel 0 with these point,. together on the usual mollview projection. How can I do this?

To be more definite, within matplotlib I can plot a point on a mollview projection like so:
```
import matplotlib.pyplot as plt
import healpy as hp
fig = plt.figure()
ax = fig.add_subplot(111, projection='mollweide')
theta, phi = hp.pix2ang(nside=1, ipix=0, nest=True)
ax.scatter(theta, phi)
```
```
import matplotlib.pyplot as plt
import healpy as hp
fig = plt.figure()
ax = fig.add_subplot(111, projection='mollweide')
theta, phi = hp.pix2ang(nside=1, ipix=0, nest=True)
ax.scatter(theta, phi) # Bu I would put in a sequence of points
```
```
import healpy as hp
import numpy as np
mymap = np.ones(hp.nside2npix(1))* hp.UNSEEN
mymap[0] = 1.
hp.mollview(mymap, nest=True )
```
which probably use different units/axes. What I want to show is the point and the healpixel together to visually show that such a point falls on the healpixel (I do not particularly care what units, as long as I know them). I am sure this is something that can be done fairly easily and it is perhaps even obvious and I am missing it

I tried your two suggestions:
`plt.gcf()` does return a figure, but it was unclear what to do with it. If I tried to save it to a file using the usual matplotlib savefig, it saved an empty file.

`return_projected_map` returns a numpy masked array. Again it is unclear how to use this to plot up the healpixels.
